pkg/ir/spec.go

```diff
@@ -125,28 +145,61 @@ type (
 		VPC            ID
 	}
 
-	SubnetSegmentDetails struct {
-		Subnets         []ID
-		OverlappingVPCs []ID
+	SegmentDetails struct {
+		Elements []ID
 	}
 
 	CidrSegmentDetails struct {
 		Cidrs            *netset.IPBlock
 		ContainedSubnets []ID
-		OverlappingVPCs  []ID
 	}
 
 	ExternalDetails struct {
 		ExternalAddrs *netset.IPBlock
 	}
 
+	Reader interface {
+		ReadSpec(filename string, defs *ConfigDefs) (*Spec, error)
+	}
+
 	Named interface {
 		Name() string
 	}
 
 	NWResource interface {
 		Address() *netset.IPBlock
 	}
+
+	INWResource interface {
```

Comment:
What's INW? Not a standard acronym. Needs a comment.
Also, do we really need both `NWResource` and `INWResource`?

Reply:
internal network resource
we need both `NWResource` and `INWResource` because not every `NWResource` is in a single subnet (e.g. VSI)

Reply:
How about `SubSubnetResource`?
I cannot find a name which will not require a comment, so please also add a comment.

Reply:
done